backends/tfhe-cuda-backend/cuda/src/polynomial/parameters.cuh

```diff
@@ -1,7 +1,17 @@
 #ifndef CUDA_PARAMETERS_CUH
 #define CUDA_PARAMETERS_CUH
 
-constexpr int log2(int n) { return (n <= 2) ? 1 : 1 + log2(n / 2); }
+#include <cstdint>
+
+// If decide to support something else than 32 and 64 bits, this method will
+// need to be adjusted
+template <typename T> constexpr unsigned log2_int(T x) {
+  if constexpr (sizeof(T) <= sizeof(uint32_t)) {
+    return (unsigned)(8 * sizeof(uint32_t) - __builtin_clz(x) - 1);
+  } else if constexpr (sizeof(T) <= sizeof(uint64_t)) {
```

Comment:
Why not take strict equality (as a uint16_t would not get the right value I think) and the last else branch have a const assert with an error message ?